### ISSUE TITLE
Can't create user through rake

### ISSUE THREAD

**Issue by p-galligan** [NCSU-Libraries/circa#4]
When running `bundle exec rake users:create_admin` I receive the following errors:

`Patricks-MacBook-Pro:circa patrick$ bundle exec rake users:create_admin
rake aborted!
RSolr::Error::ConnectionRefused: Connection refused - {:data=>"[{\"id\":\"user_1\",\"record_type\":\"user\",\"record_id\":1,\"data\":\"{\\\"id\\\":1,\\\"email\\\":\\\"admin@circa\\\",\\\"created_at\\\":\\\"2017-11-14T10:54:12-05:00\\\",\\\"updated_at\\\":\\\"2017-11-14T10:54:12-05:00\\\",\\\"position\\\":null,\\\"affiliation\\\":null,\\\"first_name\\\":null,\\\"last_name\\\":null,\\\"display_name\\\":\\\" \\\",\\\"agreement_confirmed_at\\\":null,\\\"patron_type_id\\\":null,\\\"user_role_id\\\":2,\\\"patron_type\\\":null,\\\"roles\\\":[\\\"admin\\\",\\\"staff\\\",\\\"patron\\\"],\\\"agreement_confirmed\\\":false}\",\"created_at\":\"2017-11-14T15:54:12Z\",\"updated_at\":\"2017-11-14T15:54:12Z\",\"email\":\"admin@circa\",\"user_role_id\":2,\"agreement_confirmed\":false,\"roles\":[\"admin\",\"staff\",\"patron\"]}]", :headers=>{"Content-Type"=>"application/json"}, :method=>:post, :params=>{:wt=>:json}, :query=>"wt=json", :path=>"update", :uri=>#<URI::HTTP http:/update?wt=json>}
(eval):2:in `post'
/Users/patrick/Desktop/circa/circa/app/models/search_index.rb:16:in `update_record'
/Users/patrick/Desktop/circa/circa/app/models/concerns/solr_doc.rb:246:in `update_index'
/Users/patrick/Desktop/circa/circa/lib/tasks/users.rake:9:in `block (2 levels) in <top (required)>'
/Users/patrick/.rbenv/versions/2.4.0/bin/bundle:22:in `load'
/Users/patrick/.rbenv/versions/2.4.0/bin/bundle:22:in `<main>'
Faraday::ConnectionFailed: Failed to open TCP connection to :80 (Connection refused - connect(2) for nil port 80)
(eval):2:in `post'
/Users/patrick/Desktop/circa/circa/app/models/search_index.rb:16:in `update_record'
/Users/patrick/Desktop/circa/circa/app/models/concerns/solr_doc.rb:246:in `update_index'
/Users/patrick/Desktop/circa/circa/lib/tasks/users.rake:9:in `block (2 levels) in <top (required)>'
/Users/patrick/.rbenv/versions/2.4.0/bin/bundle:22:in `load'
/Users/patrick/.rbenv/versions/2.4.0/bin/bundle:22:in `<main>'
Errno::ECONNREFUSED: Failed to open TCP connection to :80 (Connection refused - connect(2) for nil port 80)
(eval):2:in `post'
/Users/patrick/Desktop/circa/circa/app/models/search_index.rb:16:in `update_record'
/Users/patrick/Desktop/circa/circa/app/models/concerns/solr_doc.rb:246:in `update_index'
/Users/patrick/Desktop/circa/circa/lib/tasks/users.rake:9:in `block (2 levels) in <top (required)>'
/Users/patrick/.rbenv/versions/2.4.0/bin/bundle:22:in `load'
/Users/patrick/.rbenv/versions/2.4.0/bin/bundle:22:in `<main>'
Errno::ECONNREFUSED: Connection refused - connect(2) for nil port 80
(eval):2:in `post'
/Users/patrick/Desktop/circa/circa/app/models/search_index.rb:16:in `update_record'
/Users/patrick/Desktop/circa/circa/app/models/concerns/solr_doc.rb:246:in `update_index'
/Users/patrick/Desktop/circa/circa/lib/tasks/users.rake:9:in `block (2 levels) in <top (required)>'
/Users/patrick/.rbenv/versions/2.4.0/bin/bundle:22:in `load'
/Users/patrick/.rbenv/versions/2.4.0/bin/bundle:22:in `<main>'
Tasks: TOP => users:create_admin
(See full trace by running task with --trace)`

**Comment by trevorthornton** [comment]
Default values were not set, despite what the README said. I've set them now. Pull the changes and try again. Be sure that Solr is running and the core is added before running the rake task - after the user is created it will attempt to add it to the index and will fail if it can't reach Solr.

**Comment by trevorthornton** [comment]
@p-galligan - After thinking this through a bit more I realized the problem was really with the instructions. I've removed references to defaults in the configuration sections as they were misleading. There are default values in the example files, which is what the instructions were referring to. Trying to enforce defaults in the application code was causing some unforeseen problems, so the solution is really to follow the instructions for specifying appropriate values in the .yml files.

Let me know if you're still having this problem, and ensure that your config settings are correct in config/solr.yml and that Solr is running and Circa core is loaded.
